Shopify/dawn
# Remove empty space from inventory status

### PR Summary: 

Removes empty space created when the inventory status block is present, but a product has untracked inventory. 

### Why are these changes introduced?

Fixes Shopify/temp-project-mover-Archetypically-20250409144733#155

### What approach did you take?

I removed the CSS that was preventing this space from collapsing. This was initially there to present layout shift, but @vernalkick and I feel that the layout shift is a reasonable tradeoff in this case. 

~I also added some info text as recommended by @sampede. This could use some wordsmithing if we'd like to include it (cc @katycobb!).~ _Edit: We decided to remove this for now](https://github.com/Shopify/dawn/pull/3151#issuecomment-1854489466)._ 👍 

When the space collapses, it would be great to animate this so it's not so sudden. But I think that'll be way easier to accomplish with view transitions, so I've left it out for now. 

### Visual impact on existing themes

Before:


https://github.com/Shopify/dawn/assets/1202812/7f27e5f4-5020-4a24-9b46-e26f3d3ccb46


After:


https://github.com/Shopify/dawn/assets/1202812/dab43f57-8d75-4a00-90eb-c0c128f65862



### Testing steps/scenarios

1. Turn off track inventory on product/variant page and click save
2. Open the editor
3. Open the product template and view the product that you aren't tracking inventory for
4. Add the Inventory Status block, note that it takes up no space. 
5. If the product has another variant that _does_ have inventory tracked, note that there's a layout shift when switching between the two variants.

### Demo links

- [Editor](https://os2-demo.myshopify.com/admin/themes/162667528214/editor?previewPath=%2Fproducts%2F3-variants%3Fview%3D__DEFAULT__)

### Checklist
- [] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
